octobot_trading/util/test_tools/spot_rest_exchange_test_tools.py

```diff
@@ -0,0 +1,136 @@
+import octobot_trading.exchange_channel as exchanges_channel
+import octobot_trading.personal_data as trading_personal_data
+import octobot_trading.util.test_tools.exchanges_test_tools as exchanges_test_tools
+
+
+class SpotRestExchangeTests:
+    SUBSCRIBED_CHANNELS = [channels_name.OctoBotTradingChannelsName.BALANCE_CHANNEL.value,
+                           channels_name.OctoBotTradingChannelsName.TRADES_CHANNEL.value,
+                           channels_name.OctoBotTradingChannelsName.ORDERS_CHANNEL.value]
+
+    def __init__(self, exchange_name, config):
+        self.config = config
+        self.exchange_name = exchange_name
+
+        self.exchange_manager_instance = None
+
+        self.channel_callbacks_triggered = {}
+        self.expected_crypto_in_balance = []
+
+    async def initialize(self):
+        self.exchange_manager_instance = await exchanges_test_tools.create_test_exchange_manager(
+            config=self.config, exchange_name=self.exchange_name,
+            rest_only=True, is_spot_only=True, is_sandboxed=True)
+        await self.exchange_manager_instance.trader.initialize()
+
+    async def run(self, symbol):
+        await self.subscribe_personal_channels()
+        await self.exchange_manager_instance.trader.cancel_all_open_orders()
+        await self.test_orders(symbol)
+
+    async def stop(self):
+        await exchanges_test_tools.stop_test_exchange_manager(self.exchange_manager_instance)
+
+    async def test_orders(self, symbol):
+        mark_price = await self.exchange_manager_instance.exchange_symbols_data.get_exchange_symbol_data(symbol) \
+            .prices_manager.get_mark_price(timeout=60)
+        await self._create_order(symbol,
+                                 trading_enums.TraderOrderType.SELL_LIMIT,
+                                 mark_price*1.1,
+                                 0.001,
+                                 mark_price*1.1)
+        await asyncio.sleep(7)
+        assert (await self._get_exchange_orders_count(symbol)) == 1
+        open_orders = self._get_orders_manager_open_orders(symbol)
+        assert len(open_orders) == 1
+        await self.exchange_manager_instance.trader.cancel_order_with_id(open_orders[-1].order_id)
+        await asyncio.sleep(7)
+        assert (await self._get_exchange_orders_count(symbol)) == 0
+        open_orders = self._get_orders_manager_open_orders(symbol)
+        assert len(open_orders) == 0
+
+    async def _get_exchange_orders_count(self, symbol):
+        return len(await self.exchange_manager_instance.exchange.get_open_orders(symbol=symbol))
+
+    def _get_orders_manager_open_orders(self, symbol):
+        return self.exchange_manager_instance.exchange_personal_data.orders_manager.get_open_orders(symbol)
+
+    async def test_all_callback_triggered(self):
+        for channel in self.SUBSCRIBED_CHANNELS:
+            assert channel in self.channel_callbacks_triggered
+            assert self.channel_callbacks_triggered[channel]
```

Comment:
I guess we can replace those 2 lines by:
`assert self.channel_callbacks_triggered.get(channel, False)`